src/components/ProductCards.js/ProductCard.js

```diff
@@ -1,28 +1,46 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { BsCart4 } from 'react-icons/bs';
-import './ProductCard.css';
 import { Link } from 'react-router-dom';
+import './ProductCard.css';
+
+const ProductCard = ({ id, image, name, price, quantity, onDelete }) => {
+  const [isModalVisible, setIsModalVisible] = useState(false);
+
+  const showModal = () => {
```

Comment:
You can also create toggleModal that will change state to opposite. 

`const toggleModal = () => {
      setIsModalVisible(prev => !prev)
}`